packages/quantic/force-app/main/default/lwc/quanticDidYouMean/quanticDidYouMean.js

```diff
@@ -0,0 +1,70 @@
+import { api, LightningElement, track } from 'lwc';
+import { initializeWithHeadless, registerComponentForInit } from 'c/quanticHeadlessLoader';
+
+export default class QuanticDidYouMean extends LightningElement {
+    @track state = {};
+    /** @type {()=> void} */
+    unsubscribe;
+    /**
+     * @type {Boolean}
+     */
+    @track hasQueryCorrection;
+
+    /**
+     * @type {Boolean}
+     */
+    @track wasAutomaticallyCorrected;
+
+    /** @type {import("coveo").DidYouMean} */
+    didYouMean;
+    
+    /**
+     * @type {string}
+     */
+    @api engineId;
+
+    connectedCallback() {
+        registerComponentForInit(this, this.engineId);
+    }
+    
+    renderedCallback() {
+        initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+    }
+
+    /**
+     * @param {import("coveo").SearchEngine} engine
+     */
+    @api
+    initialize(engine) {
+        this.didYouMean = CoveoHeadless.buildDidYouMean(engine);
+        this.unsubscribe = this.didYouMean.subscribe(() => this.updateState());
+        
+    }
+
+    disconnectedCallback() {
+        if (this.unsubscribe) {
+          this.unsubscribe();
+        }
+      }
+
+    updateState() {
+        this.hasQueryCorrection = this.didYouMean.state.hasQueryCorrection;
+        this.wasAutomaticallyCorrected = this.didYouMean.state.wasAutomaticallyCorrected;
+    }
+
+    applyCorrection() {
+        this.didYouMean.applyCorrection();
+    }
+
+    get noResults() {
+        return this.didYouMean.state.originalQuery;
```

Comment:
I think you should not access the controller state this way, at least not outside the `updateState()` function. I fear your component might become out-of-sync with the controller as LWC refreshes your component based on the variations of the `@track` attributes, not based on the controller state changes per se.

I would suggest to change the `updateState()` function to retrieve all the fields you need from the controller state and store them into attributes marked with `@track`. This way LWC will know which parts of the component should be re-rendered when those attributes change.